src/engine/thread_pool.h

```diff
@@ -58,8 +57,8 @@ class ThreadPool {
 
     /*! \brief Signal event upon destruction, even for exceptions (RAII) */
     struct SetReadyOnDestroy {
-      explicit inline SetReadyOnDestroy(std::shared_ptr<SimpleEvent> *event)
-        : event_(*event) {
+      explicit inline SetReadyOnDestroy(const std::shared_ptr<SimpleEvent>& event)
```

Comment:
Why?

Reply:
general practice is to pass shared pointers by reference to prevent copies and in this case the pointer to shared pointer could be null and is not being checked.